pyqtgraph/opengl/items/GLTextItem.py

```diff
@@ -0,0 +1,107 @@
+            if k not in args:
+                raise Exception('Invalid keyword argument: %s (allowed arguments are %s)' % (k, str(args)))
+        for arg in args:
+            if arg in kwds:
+                value = kwds[arg]
+                if arg == 'pos':
+                    if isinstance(value, np.ndarray):
+                        if value.shape != (3,):
+                            raise Exception('"pos.shape" must be (3,).')
+                    elif isinstance(value, (tuple, list)):
+                        if len(value) != 3:
+                            raise Exception('"len(pos)" must be 3.')
+                elif arg == 'color':
+                    if isinstance(value, QtGui.QColor):
+                        pass
+                    elif isinstance(value, (tuple, list, np.ndarray)):
+                        if isinstance(value, np.ndarray):
+                            if len(value.shape) != 1:
+                                raise Exception('"color.shape" must be (3,) or (4,).')
+                        value_len = len(value)
+                        if value_len == 3:
+                            value = QtGui.QColor(value[0], value[1], value[2])
+                        elif value_len == 4:
+                            value = QtGui.QColor(value[0], value[1], value[2], value[3])
+                        else:
+                            raise Exception('"len(color)" must be 3 or 4.')
+                elif arg == 'font':
+                    if isinstance(value, QtGui.QFont) is False:
+                        raise Exception('"font" must be QFont.')
+                setattr(self, arg, value)
+        self.update()
+    
+    def paint(self):
+        if len(self.text) < 1:
+            return
+        self.setupGLState()
+
+        modelview = glGetDoublev(GL_MODELVIEW_MATRIX)
+        projection = glGetDoublev(GL_PROJECTION_MATRIX)
+        viewport = glGetIntegerv(GL_VIEWPORT)
+
+        text_pos = self.__project(self.pos, modelview, projection, viewport)
+        text_pos[1] = viewport[3] - text_pos[1]
+
+        text_pos /= self.view().devicePixelRatio()
+        
+        painter = QtGui.QPainter(self.view())
+        painter.setPen(self.color)
+        painter.setFont(self.font)
+        painter.setRenderHints(QtGui.QPainter.Antialiasing | QtGui.QPainter.TextAntialiasing)
+        painter.drawText(text_pos[0], text_pos[1], self.text)
+        painter.end()
+    
+    def __project(self, obj_pos, modelview, projection, viewport):
+        obj_vec = np.append(np.array(obj_pos), [1.0])
+
+        view_vec = np.matmul(modelview.T, obj_vec)
+        proj_vec = np.matmul(projection.T, view_vec)
+
+        if proj_vec[3] == 0.0:
+            return
+        
+        proj_vec[0:3] /= proj_vec[3]
+
+        return np.array([
+            viewport[0] + (1.0 + proj_vec[0]) * viewport[2] / 2.0,
+            viewport[1] + (1.0 + proj_vec[1]) * viewport[3] / 2.0,
+            (1.0 + proj_vec[2]) / 2.0
+        ])
+    
```

Comment:
Looks like github is complaining about no newline at the end of the file, can you verify that?

Reply:
There were four ' ' in the end of the file, so I removed them. ([8946076](https://github.com/pyqtgraph/pyqtgraph/pull/1776/commits/8946076752840caf734c783e1d05c330702620c6))